templates/common/baremetal/files/baremetal-coredns.yaml

```diff
@@ -0,0 +1,93 @@
+filesystem: "root"
+mode: 0644
+path: "/etc/kubernetes/manifests/coredns.yaml"
+contents:
+  inline: |
+    kind: Pod
+    apiVersion: v1
+    metadata:
+      name: coredns
+      namespace: openshift-kni-infra
+      creationTimestamp:
+      deletionGracePeriodSeconds: 65
+      labels:
+        app: kni-infra-mdns
+    spec:
+      volumes:
+      - name: resource-dir
+        hostPath:
+          path: "/etc/kubernetes/static-pod-resources/coredns"
+      - name: kubeconfig
+        hostPath:
+          path: "/etc/kubernetes/kubeconfig"
+      - name: conf-dir
+        empty-dir: {}
+      initContainers:
+      - name: clusterrc-generation
+        image: quay.io/openshift-metalkube/kubeconfig-extractor:latest
+        command:
+        - "/usr/bin/kubeconfig-extractor"
+        args:
+        - "/etc/kubernetes/kubeconfig"
+        - "/etc/kubernetes/static-pod-resources/clusterrc"
+        resources: {}
+        volumeMounts:
+        - name: resource-dir
+          mountPath: "/etc/kubernetes/static-pod-resources"
+        - name: kubeconfig
+          mountPath: "/etc/kubernetes/kubeconfig"
+      - name: render-corefile
+        image: quay.io/openshift/origin-node:latest
+        command:
+        - "/bin/bash"
+        - "-c"
+        - |
+          #/bin/bash
+          set -ex
+    
+          source /etc/kubernetes/static-pod-resources/clusterrc
+          CLUSTER_NAME="$NAME"
+          export CLUSTER_NAME
+          export DOMAIN
+          export NAME
+          # This should eventually be looked up from the config
+          API_VIP="$(dig +noall +answer "api.${DOMAIN}" | awk '{print $NF}')"
+          export API_VIP
+          /usr/libexec/platform-python -c "from __future__ import print_function
+          import os
+          with open('/etc/kubernetes/static-pod-resources/Corefile.template', 'r') as f:
+              content = f.read()
+          with open('/etc/coredns/Corefile', 'w') as dest:
+              print(os.path.expandvars(content), file=dest)"
+    
+          grep -v "${API_VIP}" /etc/resolv.conf | tee /etc/coredns/resolv.conf
+        resources: {}
+        volumeMounts:
+        - name: resource-dir
+          mountPath: "/etc/kubernetes/static-pod-resources"
+        - name: conf-dir
+          mountPath: "/etc/coredns"
+        imagePullPolicy: IfNotPresent
+      containers:
+      - name: coredns
+        securityContext:
+          privileged: true
+        image: quay.io/openshift-metalkube/coredns-mdns:latest
```

Comment:
I thought I mentioned this before but am not seeing it...

There is one thing that I think blocks landing here - these images need to be part of the release image and substituted by the MCO's template machinery, just like the `etcd` and `pod` images for example.  The current state means installs on metal aren't reproducible/versioned, controlled by the CVO, etc.

You want to add them to `image-references`, and at the same time I'd probably rename them to a "dummy" value like the others, i.e. make them consistently `registry.svc.ci.openshift.org/openshift:coredns-mdns` etc.

I think it may require intervention from a release admin (Clayton?) to add external images to the payload as well.

Reply:
coredns will be there soon. Hopefully today

Reply:
holding this PR till we have images before landing this

Reply:
I think I've got all of the image references correct now. /cc @cgwalters @runcom

Reply:
@runcom I guess we are waiting for the separate approved PR to merge to remove the hold. Right?

Reply:
The image references patch got merged.